File: common/src/main/java/de/werkelmann/rekote/address/port/PortRule.java
package de.werkelmann.rekote.address.port;

import de.werkelmann.rekote.address.ConnectionRule;

public class PortRule implements ConnectionRule {
    @Override
    public boolean isValid(String address) {
        try {
            Integer portNumber = Integer.parseInt(address);
            return 0 < portNumber && portNumber < 65536;
        } catch (NumberFormatException e) {
            return false;
        }
    }
}

File: common/src/main/kotlin/de/werkelmann/rekote/address/port/PortRule.kt
package de.werkelmann.rekote.address.port

import de.werkelmann.rekote.address.ConnectionRule

class PortRule : ConnectionRule {
    override fun isValid(address: String): Boolean {
        try {
            val portNumber = Integer.parseInt(address)
            return portNumber in 1..65535
        } catch (e: NumberFormatException) {
            return false
        }

    }
}
